Std/Logic.lean

```diff
@@ -349,15 +349,18 @@ variable {p q : α → Prop} {b : Prop}
 theorem forall_imp (h : ∀ a, p a → q a) : (∀ a, p a) → ∀ a, q a :=
 fun h' a => h a (h' a)
 
+@[simp] theorem forall_exists_index {q : (∃ x, p x) → Prop} :
+    (∀ h, q h) ↔ ∀ x (h : p x), q ⟨x, h⟩ :=
+  ⟨fun h x hpx => h ⟨x, hpx⟩, fun h ⟨x, hpx⟩ => h x hpx⟩
+
 theorem Exists.imp (h : ∀ a, p a → q a) : (∃ a, p a) → ∃ a, q a
   | ⟨a, hp⟩ => ⟨a, h a hp⟩
 
 theorem Exists.imp' {β} {q : β → Prop} (f : α → β) (hpq : ∀ a, p a → q (f a)) :
     (∃ a, p a) → ∃ b, q b
   | ⟨_, hp⟩ => ⟨_, hpq _ hp⟩
 
-@[simp] theorem exists_imp : ((∃ x, p x) → b) ↔ ∀ x, p x → b :=
-  ⟨fun h x hpx => h ⟨x, hpx⟩, fun h ⟨x, hpx⟩ => h x hpx⟩
+theorem exists_imp : ((∃ x, p x) → b) ↔ ∀ x, p x → b := by simp
```

Comment:
Why no longer `@[simp]`?

Reply:
I thought that lemmas provable by `simp` are generally not supposed to be marked `simp`?

Reply:
My impression was that unless the proof that `by simp` comes up with is `same_lemma_different_name`, this can still be useful. It is putting a bunch of rewrite rules together into a compound one, a shortcut for the rewrite system. @digama0 should deny/confirm

Reply:
Just because the proof is `by simp` doesn't mean that simp can do the same proof in context. And indeed it can also be useful as a shortcut like Wojciech says.

Reply:
In this case it can be proven by `apply forall_exists_index`. That is probably why the linter was complaining. So I think I should replace it with the `apply` proof for clarity, and leave the lemma no longer `@[simp]`?